ENT-9074: Added preliminary documentation about statistical classes (#2846)
cfengine/documentation: reference/components/cf-monitord.markdown

Note: There is no way for force a refresh of the monitored data.

## Statistical Classes

`cf-monitord` automatically defines classes based on the observation of the data
is has collected. Classes defined are named for the measurement id (the promise
handle in the case of custom measurement promises) with prefixes and or suffixes
depending on the measurement.

> The following suffixes may be used when defining classes:

Author: Note: I guessed at these descriptions.

Author: And a few I didn't bother to guess at, could't see through the maths.

> * `_high` :: The last measurement was greater than the average.

Reviewer: ```suggestion
* `_high` :: The last measurement seemed high. It was greater than the average of all time and also greater than the recent average.
```

> * `_low` :: The last measurement was lower than the average.

Reviewer: ```suggestion
* `_low` :: The last measurement was low. It was lower than the average of all time and also lower than the recent average. This could indicate that the measured value is experiencing a "dip" or trending in a negative direction.
```

> * `_normal` ::

Reviewer: ```suggestion
* `_normal` :: The value was neither high nor low, (as per how those are described above).
```

> * `_ldt` ::

Reviewer: A leap (step) detected, meaning a distinct (significant) change in the average.

> * `_ldt` ::

Reviewer: ```suggestion
* `_ldt` :: A leap (step) detected, meaning a distinct (significant) change in the average.
```

> * `_dev1` :: The last measurement was at least 1 standard deviation higher than the average.
> * `_dev2` :: The last measurement was at least 2 standard deviations higher than the average. These classes are persistently defined for a number of minutes.

Reviewer: ```suggestion
* `_dev1` :: The last measurement was at least 1 standard deviation higher/lower than the average.
* `_dev2` :: The last measurement was at least 2 standard deviations higher/lower than the average. These classes are persistently defined for a number of minutes.
```

* `_anomaly` :: The last measurement was at least 3 standard deviations than the average. These classes are persistently defined for a number of minutes.

> * `_microanomaly` :: The last measurement was at least 2 standard deviations higher than the average.

Reviewer: ```suggestion
* `_microanomaly` :: A deviation of more than 2 standard deviations was detected, but the difference between the last measurement and the average was below the noise threshold. (So it could mean that this anomalous value is just due to noise).
```

* `cf_state.lmdb`
* `history.lmdb`

## Statistical Classes

`cf-monitord` automatically defines classes based on the observation of the data
is has collected. Classes defined are named for the measurement id (the promise
handle in the case of custom measurement promises) with prefixes and or suffixes
depending on the measurement.

The following suffixes may be used when defining classes:

* `_high` :: The last measurement was greater than the average.
* `_low` :: The last measurement was lower than the average.
* `_normal` ::
* `_ldt` ::
* `_dev1` :: The last measurement was at least 1 standard deviation higher than the average.
* `_dev2` :: The last measurement was at least 2 standard deviations higher than the average. These classes are persistently defined for a number of minutes.
* `_anomaly` :: The last measurement was at least 3 standard deviations than the average. These classes are persistently defined for a number of minutes.
* `_microanomaly` :: The last measurement was at least 2 standard deviations higher than the average.

The following prefixes may be used when defining classes:

> * `entropy_` ::

Author: @olehermanse did you have any description for this prefix?

https://github.com/cfengine/core/blob/master/cf-monitord/env_monitor.c#L934

Reviewer: @nickanderson No, this is one of those things I'd be in favor of removing. Reduce the scope and ensure the things we do support are working well and documented.

Author: Yeah, well, when we do that we can remove it and add history for it. I assume it's something related to how much traffic is seen on a given port.
